tests/tasks/tests_create_task_api.py

```diff
@@ -0,0 +1,271 @@
+            "description": "Test",
+            "share_type": "INVALID",
+            "share_link": "..."
+        }
+        response = auth_client.post(self.create_task_api, invalid_payload)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_fail_with_invalid_platform(self):
+        auth_client = self.create_auth_client()
+
+        # Creating Admin Member
+        group = Group.objects.get(org=self.org, name='Admin')
+        member = Member.objects.create(
+            user=self.auth_user,
+            group=group,
+            org=self.org
+        )
+
+        # Payload with INVALID social_media_platform
+        invalid_payload = {
+            "social_media_platform": "INVALID",
+            "description": "Test",
+            "share_type": "LINK",
+            "share_link": "..."
+        }
+        response = auth_client.post(self.create_task_api, invalid_payload)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_fail_with_missing_share_link(self):
+        auth_client = self.create_auth_client()
+
+        # Creating Admin Member
+        group = Group.objects.get(org=self.org, name='Admin')
+        member = Member.objects.create(
+            user=self.auth_user,
+            group=group,
+            org=self.org
+        )
+
+        # Payload with empty link
+        invalid_payload = {
+            "social_media_platform": "facebook",
+            "description": "Test",
+            "share_type": "LINK",
+            "share_link": ""
+        }
+        response = auth_client.post(self.create_task_api, invalid_payload)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_fail_with_missing_share_image(self):
+
+        auth_client = self.create_auth_client()
+
+        # Creating Admin Member
+        group = Group.objects.get(org=self.org, name='Admin')
+        member = Member.objects.create(
+            user=self.auth_user,
+            group=group,
+            org=self.org
+        )
+
+        # Payload with empty IMAGE
+        invalid_payload = {
+            "social_media_platform": "whatsapp",
+            "description": "Test",
+            "share_type": "IMG",
+            "share_img": ""
+        }
+        response = auth_client.post(self.create_task_api, invalid_payload)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_fail_with_missing_share_text(self):
+
+        auth_client = self.create_auth_client()
+
+        # Creating Admin Member
+        group = Group.objects.get(org=self.org, name='Admin')
+        member = Member.objects.create(
+            user=self.auth_user,
+            group=group,
+            org=self.org
+        )
+
+        # Payload with empty TEXT
+        invalid_payload = {
+            "social_media_platform": "whatsapp",
+            "description": "Test",
+            "share_type": "TEXT",
+            "share_img": ""
```

Comment:
### 
```suggestion
            "share_img": "",
            "share_text": ""
```

Reply:
I do it with my phone... I m really sorry 😂

Reply:
Oops 😅, Thanks! 🙌